File: micronaut-mcp-server/src/test/java/io/micronaut/mcp/server/stateless/sync/prompts/PromptsFactory.java
package io.micronaut.mcp.server.stateless.sync.prompts;

/*
//tag::fakepackage[]
package example.micronaut;

//end::fakepackage[]
 */
//tag::imports[]
import io.micronaut.context.annotation.Factory;
import io.micronaut.context.annotation.Requires;
import io.modelcontextprotocol.server.McpStatelessServerFeatures;
import io.modelcontextprotocol.spec.McpSchema;
import jakarta.inject.Singleton;

import java.util.List;

//end::imports[]

@Requires(property = "spec.name", value = "StatelessSyncPromptsTest")
//tag::clazz[]
@Factory
class PromptsFactory {
    @Singleton
    McpStatelessServerFeatures.SyncPromptSpecification prompt() {
        return new McpStatelessServerFeatures.SyncPromptSpecification(
            new McpSchema.Prompt("chess-statistics", "Displays statistics for chess games",
                List.of(new McpSchema.PromptArgument("name", "Player Name", true))), (ex, req) -> {
            Object playerNameObj = req.arguments().get("name");
            String playerName = playerNameObj != null ? playerNameObj.toString() : "";
            McpSchema.TextContent assistantContent = new McpSchema.TextContent(String.format("""
                                        You generate chess statistics for %s ....""", playerName));
            McpSchema.PromptMessage assistantMessage = new McpSchema.PromptMessage(McpSchema.Role.ASSISTANT, assistantContent);
            return new McpSchema.GetPromptResult("Chess statistics", List.of(assistantMessage), null);
        });
    }
}
//end::clazz[]

File: test-suite-docs-python/src/test/python/micronaut/mcp/docs/prompts/PromptsFactory.py
# tag::imports[]
from io.modelcontextprotocol.server import McpStatelessServerFeatures
from io.modelcontextprotocol.spec import McpSchema
from jakarta.inject import Singleton
from java.util import List
from micronaut.context.annotation import Factory, Requires
# end::imports[]


@Requires(property="spec.name", value="PromptsFactoryTest")
# tag::clazz[]
@Factory
class PromptsFactory:
    @Singleton
    def prompt(self) -> McpStatelessServerFeatures.SyncPromptSpecification:
        return McpStatelessServerFeatures.SyncPromptSpecification(
            McpSchema.Prompt("chess-statistics", "Displays statistics for chess games",
                             List.of(McpSchema.PromptArgument("name", "Player Name", True))),
            lambda ctx, req: self.get_prompt_result(req))

    def get_prompt_result(self, req: McpSchema.GetPromptRequest) -> McpSchema.GetPromptResult:
        player_name_obj = req.arguments().get("name")
        player_name = str(player_name_obj) if player_name_obj is not None else ""
        assistant_content = McpSchema.TextContent(f"You generate chess statistics for {player_name} ....")
        assistant_message = McpSchema.PromptMessage(McpSchema.Role.ASSISTANT, assistant_content)
        return McpSchema.GetPromptResult("Chess statistics", List.of(assistant_message), None)
# end::clazz[]
